app/bundles/CampaignBundle/Controller/EventController.php

```diff
@@ -246,161 +246,162 @@ public function editAction($objectId)
+                ],
+                'MATCH_ONE'
+            )
+        ) {
+            return $this->modalAccessDenied();
+        }
+
+        $type      = $event['type'];
+        $eventType = $event['eventType'];
+
+        // fire the builder event
+        $events = $this->getModel('campaign')->getEvents();
+        $form   = $this->get('form.factory')->create(
+            'campaignevent',
+            $event,
+            [
+                'action'   => $this->generateUrl('mautic_campaignevent_action', ['objectAction' => 'edit', 'objectId' => $objectId]),
+                'settings' => $events[$eventType][$type],
+            ]
+        );
+        $event['settings'] = $events[$eventType][$type];
+
+        $form->get('campaignId')->setData($campaignId);
+
+        //Check for a submitted form and process it
+        if ($method === 'POST') {
+            if (!$cancelled = $this->isFormCancelled($form)) {
+                if ($valid = $this->isFormValid($form)) {
+                    $success  = 1;
+                    $formData = $form->getData();
+                    $event    = array_merge($event, $formData);
+
+                    // Set the name to the event default if not known.
+                    if (empty($event['name'])) {
+                        $event['name'] = $this->translator->trans($event['settings']['label']);
+                    }
+
+                    $modifiedEvents[$objectId] = $event;
+
+                    // Save the modified event properties to session
+                    $session->set('mautic.campaign.'.$campaignId.'.events.modified', $modifiedEvents);
+                }
             }
+        }
 
-            //ajax only for form fields
-            if (!$type || !$this->request->isXmlHttpRequest()
-                || !$this->get('mautic.security')->isGranted(
-                    [
-                        'campaign:campaigns:edit',
-                        'campaign:campaigns:create',
-                    ],
-                    'MATCH_ONE'
-                )
-            ) {
-                return $this->modalAccessDenied();
+        $viewParams = ['type' => $type];
+
+        if ($cancelled || $valid) {
+            $closeModal = true;
+        } else {
+            $closeModal = false;
+            $formThemes = ['MauticCampaignBundle:FormTheme\Event'];
+
+            if (isset($event['settings']['formTheme'])) {
+                $formThemes[] = $event['settings']['formTheme'];
             }
 
-            //fire the builder event
-            $events = $this->getModel('campaign')->getEvents();
-            $form   = $this->get('form.factory')->create(
-                'campaignevent',
-                $event,
+            $viewParams = array_merge($viewParams, [
+                'form'             => $this->setFormTheme($form, 'MauticCampaignBundle:Campaign:index.html.php', $formThemes),
+                'eventHeader'      => $event['settings']['label'],
```

Comment:
These are already translated in the CampaignBuilderEvent when adding the action: https://github.com/mautic-internal/mautic/blob/staging.campaign-jump-to-event/app/bundles/CampaignBundle/Event/CampaignBuilderEvent.php#L239